sqlmesh/dbt/basemodel.py

```diff
@@ -236,6 +237,16 @@ def sqlmesh_model_kwargs(self, context: DbtContext) -> t.Dict[str, t.Any]:
             "jinja_macros": jinja_macros,
             "path": self.path,
             "hash_raw_query": True,
+            "pre_statements": [
+                exp
+                for hook in self.pre_hook
+                for exp in d.parse(hook.sql, default_dialect=self.model_dialect or context.dialect)
```

Comment:
Include sql_header as the first pre_hook if sql_header is not None. I merged changes today that had this behavior in Model.py, which should be here now instead.

Reply:
Note that the removed code in Model.py used `pre_hooks` var and not `self.pre_hook`

Reply:
good catch, thanks